pyro/distributions/transforms/naf.py

```diff
@@ -13,12 +13,123 @@
 eps = 1e-8
 
 
+class ELUMixin(object):
```

Comment:
nit: Instead of using a constant eps, would suggest adapting eps depending on tensor dtype as follows:

```python
def _eps(x):
    return torch.finfo(x.dtype).tiny
```

Reply:
`.tiny` is the machine epsilon for a type, which I think may be too small to prevent numerical instabilities (although I'm not entirely certain). What do you think?

Reply:
We have found it to work quite well e.g. for SigmoidTransform and many other transforms in PyTorch. If you don't need high precision and the have found 1e-8 to work well in practice, you don't have to worry about it though.

Reply:
I've replaced it with your suggestion